fdbserver/storageserver.actor.cpp

```diff
@@ -3224,6 +3287,30 @@ void StorageServerDisk::writeMutations(const VectorRef<MutationRef>& mutations,
 	}
 }
 
+ACTOR Future<int64_t> _asyncMakeVersionMutationsDurable(StorageServerDisk* self, StorageServer* data, Version startStorageVersion, Version desiredVersion, int64_t bytesLeft) {
+	state Version newOldestVersion = startStorageVersion;
+	// Write mutations to storage until we reach the desiredVersion or have written too much (bytesleft)
+	loop {
+		state bool done = self->makeVersionMutationsDurable(newOldestVersion, desiredVersion, bytesLeft);
+		// We want to forget things from these data structures atomically with changing oldestVersion (and "before", since oldestVersion.set() may trigger waiting actors)
+		// forgetVersionsBeforeAsync visibly forgets immediately (without waiting) but asynchronously frees memory.
+		Future<Void> finishedForgetting = data->mutableData().forgetVersionsBeforeAsync( newOldestVersion, TaskPriority::UpdateStorage );
+		data->oldestVersion.set( newOldestVersion );
+		wait( finishedForgetting );
+		wait( yield(TaskPriority::UpdateStorage) );
+		if (done) break;
+	}
+
+	// Set the new durable version as part of the outstanding change set, before commit
+	if (startStorageVersion != newOldestVersion)
+		self->makeVersionDurable( newOldestVersion );
+	return bytesLeft;
+}
+
+Future<int64_t> StorageServerDisk::asyncMakeVersionMutationsDurable(Version startStorageVersion, Version desiredVersion, int64_t bytesLeft) {
+	return _asyncMakeVersionMutationsDurable(this, data, startStorageVersion, desiredVersion, bytesLeft);
```

Comment:
The naming convention for actor member function implementations is generally <name>_impl throughout the codebase.

Reply:
Done.